nbconvert/filters/markdown_mistune.py

```diff
@@ -53,8 +84,17 @@ class MarkdownWithMath(mistune.Markdown):
     def __init__(self, renderer, **kwargs):
         if 'inline' not in kwargs:
             kwargs['inline'] = MathInlineLexer
+        if 'block' not in kwargs:
+            kwargs['block'] = MathBlockLexer
         super(MarkdownWithMath, self).__init__(renderer, **kwargs)
 
+    def output_block_math(self):
+        return self.renderer.block_math(self.token["text"])
```

Comment:
This should be `return self.inline(self.token["text"])` which will internally handle delegation to the different renderers.